js/models/Profile.js

```diff
@@ -0,0 +1,158 @@
+import _ from 'underscore';
+import BaseModel from './BaseModel';
+import app from '../app';
+import is from 'is_js';
+
+export default class extends BaseModel {
+  defaults() {
+    return {
+      primaryColor: '#086A9E',
+      secondaryColor: '#317DB8',
+      textColor: '#ffffff',
+      social: [],
+    };
+  }
+
+  url() {
+    // url is handled by sync, but backbone bombs if I don't have
+    // something explicitly set
+    return 'use-sync';
+  }
+
+  getColorFields() {
+    return [
+      'primaryColor',
+      'secondaryColor',
+      'textColor',
+    ];
+  }
+
+  // Ensure any colors are strings and have a leading hash.
+  standardizeColorFields(attrs = {}) {
+    const updatedAttrs = { ...attrs };
+
+    this.getColorFields().forEach((field) => {
+      if (typeof attrs[field] !== 'undefined') {
+        updatedAttrs[field] = updatedAttrs[field].toString();
+        updatedAttrs[field] = updatedAttrs[field].charAt(0) !== '#' ?
+          `#${updatedAttrs[field]}` : updatedAttrs[field];
+      }
+    });
+
+    return updatedAttrs;
+  }
+
+  set(key, val, options) {
+    // Handle both `"key", value` and `{key: value}` -style arguments.
+    let attrs;
+    let opts = options;
+
+    if (typeof key === 'object') {
+      attrs = key;
+      opts = val;
```

Comment:
This is actually code taken right from backbone source. What it's doing is allowing the function to handle 2 different signartures 1.) `set(keyString, val, options)` 2.) `set(attrObject, options`).

So... if key is passed in as an object, then options would be passed in as the second arg (val) and the line you're looking at is just setting the opts to that.